.teamcity/Gradle_Check/model/CIBuildModel.kt

```diff
@@ -238,30 +269,41 @@ data class GradleSubprojectBucket(val name: String, val subprojects: List<Gradle
         }
     }
 
-    constructor(subproject: GradleSubproject, number: Int = 1, total: Int = 1) : this(name = "${subproject.name}${if (number == 1) "" else "_$number"}", subprojects = listOf(subproject), number = number, total = total)
+    private
+    fun hasSameProperties(predicate: (GradleSubproject) -> Boolean): Boolean {
+        val count = subprojects.count(predicate)
+        return count == 0 || count == subprojects.size
+    }
 
-    // TODO: Hacky. We should really be running all the subprojects on macOS
-    // But we're restricting this to just a subset of projects for now
-    // since we only have a small pool of macOS agents
-    fun shouldBeSkipped(testCoverage: TestCoverage) = subprojects.map { it.name }.intersect(testCoverage.os.ignoredSubprojects).isNotEmpty()
+    override fun shouldBeSkipped(testCoverage: TestCoverage) = subprojects.map { it.name }.intersect(testCoverage.os.ignoredSubprojects).isNotEmpty()
```

Comment:
Shouldn't you query here if any of the subprojects should be skipped instead of repeating the condition?